File: 2017/Lecture 4/GitHubProjects/app/src/main/java/com/avast/example/android/github/di/ApplicationModule.java
package com.avast.example.android.github.di;

import javax.inject.Singleton;

import android.content.Context;

import dagger.Module;
import dagger.Provides;

/**
 * @author Lukas Prokop (prokop@avast.com)
 */
@Module
public class ApplicationModule {

    private final Context mContext;

    public ApplicationModule(Context context) {
        mContext = context;
    }

    @Provides
    @Singleton
    public Context provideAppContext() {
        return mContext;
    }
}

File: 2017/Lecture 6/App/app/src/main/java/com/avast/android/githubbrowser/di/ApplicationModule.kt
package com.avast.android.githubbrowser.di

import android.content.Context
import com.avast.android.githubbrowser.GithubApplication
import dagger.Module
import dagger.Provides
import javax.inject.Singleton

/**
 * Created by lukas on 12/21/17.
 */
@Module
class ApplicationModule(private val mApplication: GithubApplication) {

    @Provides
    @Singleton
    fun provideAppContext(): Context {
        return mApplication.applicationContext
    }
}